### PR TITLE
Add railsy at 0.1.1

### PR DESCRIPTION
**Pull request by mmkaram**
## Description of changes
added railsy package, [homepage](https://mmkaram.github.io/railsy.html)
added mmkaram to maintainer-list
added default.nix to railsy in pkgs/by-name/railsy
added required top level definitions

This is my first attempt at adding a package to nixpkgs, so please let me know if I did anything wrong

## Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by steeleduncan** [comment]
The name of the commit adding yourself to the maintainers list should be `maintainers: add mmkaram`

**Review by steeleduncan — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ra/railsy/package.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "railsy";
+  version = "0.1.1";
+
+  src = fetchFromGitHub {
+    owner = "mmkaram";
+    repo = "railsy";
+    rev = "ref/tags/v${version}";
```
Comment (steeleduncan) [inline pkgs/by-name/ra/railsy/package.nix:16]: When I tried building this, there is no such tag in the mmkaram/railsy repo, you'd need to add it, or use a commit as the `rev`

**Review by steeleduncan — CHANGES_REQUESTED** [review]

**Comment by steeleduncan** [comment]
By the way, it is not for me to decide, but this looks a lot like a hobby project, with very few commits or users. I think it is unlikely to be picked up to be merged into `nixpkgs` at this point

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -5148,6 +5148,8 @@ with pkgs;
 
   rainbowstream = with python3.pkgs; toPythonApplication rainbowstream;
 
+  railsy = callPackage ../by-name/ra/railsy { };
```
[the remark this answers is not in the document]
Comment (cu1ch3n) [reply pkgs/top-level/all-packages.nix:5151]: @steeleduncan You are right. Please remove this line.

**Comment by cu1ch3n** [comment]
Hi @mmkaram! Thanks for the contribution but I would suggest you to read the pkgs/README and search for some similar PR examples to see if something can be learned or borrowed before submitting a PR. It will save our time. As you can see, there are thousands of PRs not reviewed.

- You should NOT add include multiple random changes in one commit. You cannot have a commit saying "maintainers: add mmkaram" but you put everything unrelated there.
- You should NOT use random commit message like "Add railsy at 0.1.1"
- You should make sure that the package can be built at least on your machine
- This project by yourself is still too immature and should not be merged into nixpkgs. I would suggest you submit it to NUR https://github.com/nix-community/NUR.

**Comment by mmkaram** [comment]
Hey duncan and cu1ch3n, thank you guys for the feedback! I'll get to all these action items and check out the NUR. Have a good one!

**Pull request closed without merging**